azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -365,6 +367,10 @@ func resourceKubernetesClusterNodePoolCreate(d *schema.ResourceData, meta interf
 		profile.VnetSubnetID = utils.String(vnetSubnetID)
 	}
 
+	if upgradeSettingsRaw, ok := d.Get("upgrade_settings").([]interface{}); ok && len(upgradeSettingsRaw) > 0 {
```

Comment:
we shouldn't need this, we can always expand this, since the expand should return the default values here if this is empty:

```suggestion
	profile.UpgradeSettings = expandUpgradeSettings(d.Get("upgrade_settings").([]interface{}))
```